### ISSUE TITLE
Yabeda v0.13.0 breaks Ruby 2.7

### ISSUE THREAD

**Issue by bibendi** [yabeda-rb/yabeda#40]
Hey! [This feature](https://github.com/yabeda-rb/yabeda/commit/7df0d8eba880cd5d191f583d1112046cc856ba53#diff-3534c3e3c3a7424e2df351c68f00f10e7cb516c4f092a030d4ede643b1649de2R6) has broke Ruby 2.7

```
irb(Yabeda):001> labels
=> {:config=>"primary", :kind=>"SCHEMA", :cached=>false, :async=>true}

irb(Yabeda):002> ::Yabeda.activerecord.queries_total.tags
=> [:config, :kind, :cached, :async]

irb(Yabeda):003> ::Yabeda.activerecord.queries_total.increment(labels)
/usr/local/bundle/ruby/2.7.0/gems/yabeda-0.13.0/lib/yabeda/counter.rb:6:in `increment': unknown keywords: :config, :kind, :cached, :async (ArgumentError)

irb(Yabeda):004> ::Yabeda.activerecord.queries_total.increment(labels, by: 1)
=> 2
```

**Comment by Envek** [comment]
Ruby 2.x keyword argument handling for methods with arguments _with default values_ was an absolute mess. See https://github.com/yabeda-rb/yabeda/issues/26#issuecomment-2405100786

I shouldn't release this as is (I just didn't notice CI failure for 2.7 due to flaky specs in previous commits).

Probably I'm going to yank this release and re-release it with Ruby 2.x support dropped and minimum required Ruby bumped to 3.0

**Comment by bibendi** [comment]
Doesn't the [`ruby2_keywords`](https://devdocs.io/ruby~2.7/module#method-i-ruby2_keywords) help here?

**Comment by Envek** [comment]
Something like this would work (but will be ugly as hell):
```ruby
# def increment(tags = {}, by: 1)
def increment(*args)
 tags, by = parse_args(*args)
 # …
end

def parse_args(*args)
case args.size
when 0 # increment()
  [{}, 1]
when 1 # increment(by: 5) or increment(tags)
  if args[0].key?(:by)
    [{}, args.fetch(:by)]
  else
    [args[0], 1]
  end
when 2 # increment(tags, by: 5)
  [args[0], args[1].fetch(:by, 1)]
else
  raise ArgumentError, "wrong number of arguments (given #{args.size}, expected 0..2)"
end
end
```

**Comment by Envek** [comment]
Fix has been released in [0.13.1](https://github.com/yabeda-rb/yabeda/releases/tag/v0.13.1)
